proxy/src/coco/peer.rs

```diff
@@ -106,6 +124,71 @@ pub fn list_projects(peer: &PeerApi) -> Result<Vec<Project>, error::Error> {
         .collect()
 }
 
+/// Get all peer IDs and their branches.
+///
+/// # Errors
+///
+///   * [`error::Error::LibradLock`]
+///   * [`error::Error::Git`]
+pub fn revisions(
+    peer: &PeerApi,
+    owner: &User,
+    project_urn: &RadUrn,
+) -> Result<NonEmpty<Revision>, error::Error> {
+    let project = get_project(peer, project_urn)?;
+    let storage = peer.storage().reopen()?;
+    let repo = storage.open_repo(project.urn())?;
+
+    let (local_branches, local_tags) = with_browser(peer, &project.urn(), |browser| {
+        Ok((
+            source::branches(browser, Some(BranchType::Local))?,
+            source::tags(browser)?,
+        ))
+    })?;
+
+    let mut remotes = NonEmpty::new(Revision {
+        identity: owner.clone().into(),
+        branches: local_branches,
+        tags: local_tags,
+    });
```

Comment:
What if the current user doesn't have their own copy of the project? I would expect that unless I've called `git push` at some point in time, I don't have a publicly visible repo under my name.

Reply:
Ya, good catch! Easy to forget when all our tests involve us creating and owning the project :facepalm:

Reply:
This is done now @cloudhead :)